diff --git a/megatron/core/utils.py b/megatron/core/utils.py
index f80e3e04..e89aed19 100644
--- a/megatron/core/utils.py
+++ b/megatron/core/utils.py
@@ -62,6 +62,24 @@ def get_torch_version():
     return _torch_version
 
 
+def deprecate_inference_params(inference_context, inference_params):
+    """Print warning for deprecated `inference_params`."""
+    if inference_context is None and inference_params is not None:
+        warnings.warn(
+            "`inference_params` renamed to `inference_context`, and will be "
+            "removed in `megatron-core` 0.13."
+        )
+        return inference_params
+    return inference_context
+
+
+def is_fa_min_version(version, check_equality=True):
+    """Check if minimum version of `flash-attn` is installed."""
+    if check_equality:
+        return get_fa_version() >= PkgVersion(version)
+    return get_fa_version() > PkgVersion(version)
+
+
 def get_te_version():
     """Get TE version from __version__; if not available use pip's. Use caching."""
 
@@ -688,6 +706,26 @@ def local_multi_tensor_scale(chunk_size, noop_flag, tensor_lists, scale):
         dst.copy_(src * scale)
 
 
+class WrappedTensor:
+    """
+    A wrapper for tensors that enables caller functions to pass an indirect reference
+    to callee functions. By wrapping the tensor, the caller's direct reference is removed,
+    allowing the tensor to be garbage collected once the callee unwraps and frees it.
+    """
+
+    def __init__(self, tensor: torch.Tensor):
+        self._wrapper = [tensor]
+
+    def unwrap(self):
+        """
+        Returns the wrapped tensor while deleting the internal reference.
+        Can only be called once.
+        """
+        if len(self._wrapper) == 0:
+            raise RuntimeError(f"WrappedTensor has already been unwrapped")
+        return self._wrapper.pop(0)
+
+
 class _ValueWithRank:
     """This is an internal class, not for use outside this module
 

