diff --git a/megatron/training/checkpointing.py b/megatron/training/checkpointing.py
index 749563ca..9fe2218e 100644
--- a/megatron/training/checkpointing.py
+++ b/megatron/training/checkpointing.py
@@ -1232,6 +1232,7 @@ def load_checkpoint(model, optimizer, opt_param_scheduler, load_arg='load', stri
     set_checkpoint_version(state_dict.get('checkpoint_version', 0))
 
     # Set iteration.
+    #print(f'info+ {args.finetune} {list(state_dict.keys())}')
     if args.finetune or release:
         iteration = 0
     else:
@@ -1267,7 +1268,8 @@ def load_checkpoint(model, optimizer, opt_param_scheduler, load_arg='load', stri
     strict = False if args.retro_add_retriever else strict
     if not skip_load_to_model_and_opt:
         if len(ddp_model) == 1:
-            ddp_model[0].load_state_dict(state_dict['model'], strict=strict)
+
+            ddp_model[0].load_state_dict(state_dict['model'], strict=False)
         else:
             for i in range(len(ddp_model)):
                 mpu.set_virtual_pipeline_model_parallel_rank(i)
@@ -1283,6 +1285,7 @@ def load_checkpoint(model, optimizer, opt_param_scheduler, load_arg='load', stri
         try:
             # Load state dict.
             if not skip_load_to_model_and_opt and optimizer is not None and not optimizer.is_stub_optimizer:
+                # print(state_dict.keys())
                 optimizer.load_state_dict(state_dict['optimizer'])
 
             # Load distributed optimizer's custom parameter state.
@@ -1380,7 +1383,6 @@ def load_checkpoint(model, optimizer, opt_param_scheduler, load_arg='load', stri
         # Notify FT that a checkpoint was loaded.
         is_local_chkpt = (ckpt_type == CheckpointType.LOCAL)
         ft_integration.on_checkpoint_loaded(is_local_chkpt=is_local_chkpt)
-
     return iteration, num_floating_point_operations_so_far
 
 

